### PR TITLE
Will Jones Refactored TacoBells Project MKI

### PR DESCRIPTION
**Pull request by JohnWilliamJonesIII**
Hey Chip!

I took your critique and tried my hand at refactoring my TacoBells project before it became too far away in my mind.

I made the tests for the ShouldFailParse method actually functional and all working, though I couldn't get the > or < than latitude and longitude edge cases working, which is something I'll hopefully tackle in the next version.

I also fixed the main TacoParser that wouldn't work if the ShouldFailParse method was actually implemented thing and I removed all the empty spaces, though I think that I could have put some more commented lines to break up and label everything.

Also Kris Dixon told me about the final two answers for the project while I was stuck in class one day lol, IIRC I just plugged them in to make sure I was on the right track if I became stuck.

If there's anything I could make more readable, or account for more edge cases or something, I'd love to hear your recommendations.

Thanks Chip!

### CODE REVIEW

File: LoggingKata/TacoParser.cs [hunk 1]
```diff
@@ -6,13 +6,37 @@
...
-            // Do not fail if one record parsing fails, return null
-            return null; // TODO Implement
+            TacoBell newTacoBell = new TacoBell();
+            double Latitude;
+            double Longitude;
+            string string_Latitude = cells[0];
+            string string_Longitude = cells[1];
+            newTacoBell.Name = cells[2];
+            bool latitiude_string_to_double = double.TryParse(string_Latitude, out Latitude);
+            bool longitude_string_to_double = double.TryParse(string_Longitude, out Longitude);
```
Comment (chipelmer) [inline LoggingKata/TacoParser.cs:29]: For these TryParse lines, consider:
```CS
bool latitiude_string_to_double = double.TryParse(string_Latitude, out double Latitude);
bool longitude_string_to_double = double.TryParse(string_Longitude, out double Longitude);
```
and removing the declaration lines 23-24

File: LoggingKata/Program.cs [hunk 2]
```diff
@@ -12,18 +12,37 @@ class Program
...
+                ITrackable locA = locations[a];
+                GeoCoordinate corA = new GeoCoordinate(locA.Location.Latitude, locA.Location.Longitude);
+                for (int b = 0; b < locations.Length; b++)
+                {   // Goes through the .csv string to find the Longitude + Latitude for the 2nd Taco Bell
+                    ITrackable locB = locations[b];
+                    GeoCoordinate corB = new GeoCoordinate(locB.Location.Latitude, locB.Location.Longitude);
+                    double currentDistance = corA.GetDistanceTo(corB);
+                    if (currentDistance >= distanceBetweenTacobells)
+                    {
+                        distanceBetweenTacobells = corA.GetDistanceTo(corB);
```
Comment (chipelmer) [inline LoggingKata/Program.cs:38]: No need to re-calculate this. `distanceBetweenTacobells = currentDistance;`

File: LoggingKata/TacoParser.cs [hunk 3]
```diff
@@ -6,13 +6,37 @@
...
+                return null;
+            }
             logger.LogInfo("Begin parsing");
-
-            // Do not fail if one record parsing fails, return null
-            return null; // TODO Implement
+            TacoBell newTacoBell = new TacoBell();
+            double Latitude;
+            double Longitude;
+            string string_Latitude = cells[0];
```
Comment (chipelmer) [inline LoggingKata/TacoParser.cs:25]: Naming convention-wise, C# tends to use pascal or camel case, with Python usually using snake case. Consider naming these with camel as `stringLatitude` or Hungarian notation as `strLatitude`

**Review by chipelmer — APPROVED** [review]
First of all, I think you're the only person that has taken the time and effort to refactor this project, so props to that 👍 
Second, this looks very good
